python/setup.py

```diff
@@ -227,7 +227,10 @@ def get_packages(self):
 if setup_spec.type == SetupType.RAY:
     pandas_dep = "pandas >= 1.3"
     numpy_dep = "numpy >= 1.20"
-    pyarrow_dep = "pyarrow >= 6.0.1,<18"
+    pyarrow_deps = [
+        "pyarrow >= 6.0.1,<18; sys_platform == 'darwin'",
```

Comment:
just for our future selves, pyarrow 18 starts releasing for macos 12, where ray is still releasing for macos 10.

we are capping as we are even already thinking about stop supporting intel silicon.. the world of macos is moving to apple silicon.

Reply:
> pyarrow 18 starts releasing for macos 12, where ray is still releasing for macos 10.

thanks @can-anyscale for discovering this.

Reply:
not sure, but maybe limit to x86 too? I think the issue that blocks the release tests is intel silicon only.

Reply:
> but maybe limit to x86 too

I am also kind of okay with its current form.. but limiting `<18` to x86 arch feels better. I am okay with just firing all the intel silicon users, but I am a bit worry that some macos apple silicon ray user in the future want to use with pyarrow 18+.

Reply:
> limiting <18 to x86 arch feels better

something like this @aslonnie ?

```
pyarrow_deps = [
        "pyarrow >= 6.0.1,<18; sys_platform == 'darwin' and platform_machine == 'x86_64'",
        "pyarrow >= 6.0.1; sys_platform != 'darwin' or platform_machine != 'x86_64'",
    ]
```

Reply:
yeah exactly.

I personally would write:

````
pyarrow >= 6.0.1
pyarrow <18; sys_platform == 'darwin' and platform_machine == 'x86_64'
````

Reply:
logically equivalent, but might reads clearer on the intention.

Reply:
much cleaner, thanks!

Reply:
don't forget to sync to the other place (`python/requirements.txt`) too.